### ISSUE TITLE
 Trying to get property 'base' of non-object in .../Core/Admin_Bar/Admin_Bar.php on line 147

### ISSUE THREAD

**Issue by RealBenihime** [google/site-kit-wp#403]
## Bug Description

So far I haven't picked up any issues with operations, however, the error does show up on the frontend only of the site.

Notice: Trying to get property 'base' of non-object in /.../wp-content/plugins/google-site-kit/includes/Core/Admin_Bar/Admin_Bar.php on line 147

## Steps to reproduce

1. Install and Activate Google Site Kit Plugin
2. Complete Setup Plugin
3. Navigate to frontend (Debug must be set to True on wp-config.php)

## Additional Context

 - PHP Version: 7.2.19-0ubuntu0.18.04.1
 - OS: [e.g. iOS] (Client)Windows 10 | (Server)Linux
 - Browser: [e.g. chrome, safari]Chrome
 - Plugin Version: [e.g. 22] 1.0.0-beta.1.0.3
 - Device: [e.g. iPhone6] Dell Inspiron 15 7000

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* No PHP error notice should appear from the Site Kit admin bar when browsing the frontend.
* The plugin should not rely on a "current screen" being available just because the function is available.

## Implementation Brief

The logic for determining if the Admin Bar should display for a particular screen (i.e. wp-admin) is convoluted and not easy to reason about. Ensuring that a `WP_Screen` object is available removes the need to check for the availability of specific properties as well.

* Extract this logic from `\Google\Site_Kit\Core\Admin_Bar\Admin_Bar::is_active` to a new private `is_admin_post_screen` method on `Admin_Bar` 
```php
        if ( ! is_user_logged_in() || ! is_admin_bar_showing() ) {
            return false;
        }

        // condition was here
        
        if ( $this->is_admin_post_screen() ) {
            $post = get_post();
        } else {
            $post = get_queried_object();
        }
```

```php
private function is_admin_post_screen() {
    $current_screen = function_exists( 'get_current_screen' ) ? get_current_screen() : false;

    // No screen context available
    if ( ! $current_screen instanceof \WP_Screen ) {
        return false;
    }

    // Only show for post screens
    if ( 'post' !== $current_screen->base ) {
        return false;
    }

    // Don't show for new post screen
    if ( 'add' === $current_screen->action ) {
        return false;
    }

    return true;
}
```

## Changelog entry

* Fix potential PHP notice when no current WordPress admin screen is present.

**Comment by swissspidy** [comment]
Thanks for your report!

Looks like there is a missing `$current_screen === null` check here:

https://github.com/google/site-kit-wp/blob/a0e6e72ecac55ac8adf44f7e16b2a8a32fd4a417/includes/Core/Admin_Bar/Admin_Bar.php#L143-L150

**Comment by RealBenihime** [comment]
Thanks @swissspidy 

I've added the null check and returned false if $current_screen is null, that did the trick.
But I think I'll keep this ticket open until the fix is applied in the next build.

**Comment by swissspidy** [comment]
Yeah, please don't close :-) The intention was not to give you a hint for a fix, but to fix this properly in the plugin.

**Comment by Reaper1242** [comment]
Hi,
Just wanted to add that I'm seeing the same.

**Comment by aaemnnosttv** [comment]
Rather than checking for various falsey values we should instead check that `$current_screen` is indeed an instance of `WP_Screen`.

**Comment by felixarntz** [comment]
@aaemnnosttv IB looks good, with two recommendations:
1. We could rename `is_active_for_current_screen()` to `is_admin_post_screen()`, because that's what it checks.
2. We can probably remove the `is_admin()` check before calling `is_active_for_current_screen()` / `is_admin_post_screen()` because the function won't exist outside of `is_admin()` anyway.

**Comment by aaemnnosttv** [comment]
IB updated with your recommendations @felixarntz

**Comment by swissspidy** [comment]
I assume it was some conflict with another plugin

**Comment by swissspidy** [comment]
The latter.

@RealBenihime maybe you can shed some light on this? Which other plugins are you using?

**Comment by RealBenihime** [comment]
@swissspidy Apparently it is a plugin conflict and I narrowed it down to this plugin (WP File Manager) https://wordpress.org/plugins/wp-file-manager I've confirmed it on 2 sites.

**Comment by swissspidy** [comment]
Awesome, thanks for checking!

@tofumatt Can you try it with the WP File Manager plugin installed?

**Comment by circlecube** [comment]
I've tried to reproduce this error without the plugin to no success. I also tried with the plugin installed and still no success. I tried with all the latest default themes (17 18 & 19). Any other specific instruction to reproduce @RealBenihime?

**Comment by RealBenihime** [comment]
@circlecube I couldn't think of any other way to do this but here is a YT video of a screen recording reproducing the error [https://youtu.be/bSkwHkbwkew](https://youtu.be/bSkwHkbwkew) and attached is the system info that was generated during the recording
[system-info-cva.ukzn.ac.za-28-08-2019.txt](https://github.com/google/site-kit-wp/files/3550783/system-info-cva.ukzn.ac.za-28-08-2019.txt)

**Comment by swissspidy** [comment]
@RealBenihime Here's a current build of the plugin that includes this bugfix:

[google-site-kit.zip](https://github.com/google/site-kit-wp/files/3550890/google-site-kit.zip)

Could you please give it a try and let us know if it resolves the issue?

**Comment by RealBenihime** [comment]
@swissspidy It does resolve the issue BUT, it also removed the stats on the Adminbar, only thing displayed now is the "Stats for: [page_name]" and "More Details" link

**Comment by aaemnnosttv** [comment]
@swissspidy @felixarntz The only thing we changed related to this issue specifically was whether or not the Site Kit admin bar item is displayed, where the screen object was accessed that raised the notice. It may be possible a change was made elsewhere that introduced a regression related to what is displayed but that should likely be a separate issue.

**Comment by circlecube** [comment]
I was able to reproduce this issue with current develop branch. Only when I also installed the Elementor plugin though. It only occurs if you have both Elementor AND WP File Manager plugin installed and active along with Site Kit. The Site Kit dropdown in the menu still functions as expected, but at the top of the page I see the notice:
`Notice: Trying to get property 'base' of non-object in /wp-content/plugins/google-site-kit/includes/Core/Admin_Bar/Admin_Bar.php on line 147`
![Screenshot 2019-08-28 16 08 11](https://user-images.githubusercontent.com/698049/63868393-57709800-c9ae-11e9-88c2-46ca2c721c50.png)
